packages/hash/api/src/db/adapter.ts

```diff
@@ -157,6 +157,26 @@ export type Graph = {
   }[];
 };
 
+export type DbBlockProperties = {
+  entityId: string;
+  accountId: string;
+  componentId?: string;
```

Comment:
`BlockProperties.componentId` doesn't seem to be optional in the generated API types which these were based off in the DbTypes file: [`Omit<BlockProperties, "entity">`.](https://github.com/hashintel/hash/blob/a5a385fe16650b5dd6a715bafdf26ffd718f5d6c/packages/hash/api/src/types/dbTypes.ts#L7)

Should this be non-optional?

Reply:
Thanks for reminding me, I wanted to ask people about this but forgot. The reason I made it optional is I found some code in the resolvers that was creating block properties objects without passing any `componentId`.

<img width="671" alt="image" src="https://user-images.githubusercontent.com/32243840/150970662-b0c6fae6-ae9b-4dfa-a9bc-0c871b3516b7.png">

I wasn't sure if this was intentional or a mistake. Now I'm just realising that it might match the fact that the old `DbPageProperties` type was overriding the `contents` property and only taking `entityId` and `accountId`, which might explain why it was fine before.

```ts
export type DbPageProperties = Omit<Page["properties"], "contents"> & {
   contents: {
     accountId: string;
     entityId: string;
   }[];
 };
```

I have no idea why it's done like that though

Reply:
Looks like the `componentId` could be passed to that `insertBlock` function. Not sure what is the right thing to do here, though. Perhaps @nathggns knows since this touches the save functionality of the frontend

Reply:
Yeah, I'm not sure if it's intentional or not to not have `componentId` in the db contents

Reply:
The issue here is that `DbPageProperties.contents` should just be:
```typescript
{
   accountId: string;
   entityId: string;
}
```

These are the `accountId` and `entityId` linking to a separate db record, the `DbBlock`. So there's no `componentId` because that lives on the linked record, not the Page record.

The `DbBlock` has the properties as described, except with `componentId` as required.